security/pkg/platform/client.go

```diff
@@ -41,12 +41,18 @@ func NewClient(platform, rootCertFile, keyFile, certChainFile, caAddr string) (C
 	case "onprem":
 		return NewOnPremClientImpl(rootCertFile, keyFile, certChainFile)
 	case "gcp":
-		return NewGcpClientImpl(rootCertFile, caAddr), nil
+		// Temporarily disable GCP credential authentication on CSR API.
+		// [TODO](myidpt): enable when the Citadel authz can work correctly.
+		return nil, fmt.Errorf("GCP credential authentication in CSR API is disabled")
 	case "aws":
-		return NewAwsClientImpl(rootCertFile), nil
+		// Temporarily disable AWS credential authentication on CSR API.
+		// [TODO](myidpt): enable when the Citadel authz can work correctly.
+		return nil, fmt.Errorf("AWS credential authentication in CSR API is disabled")
 	case "unspecified":
 		if metadata.OnGCE() {
-			return NewGcpClientImpl(rootCertFile, caAddr), nil
+			// Temporarily disable GCP credential authentication on CSR API.
```

Comment:
Please return the 'onPrem' instead of error if unspecified, it is safer.

Reply:
Done.